Sources/UnidirectionalBinding.swift

```diff
@@ -11,167 +11,61 @@ precedencegroup BindingPrecedence {
 
 infix operator <~ : BindingPrecedence
 
-/// Describes a target to which can be bound.
-public protocol BindingTargetProtocol: class {
+/// Describes a source which can be bound.
+public protocol BindingSourceProtocol {
 	associatedtype Value
-
-	/// The lifetime of `self`. The binding operators use this to determine when
-	/// the binding should be teared down.
-	var lifetime: Lifetime { get }
-
-	/// Consume a value from the binding.
-	func consume(_ value: Value)
-
-	/// Binds a signal to a target, updating the target's value to the latest
-	/// value sent by the signal.
-	///
-	/// - note: The binding will automatically terminate when the target is
-	///         deinitialized, or when the signal sends a `completed` event.
-	///
-	/// ````
-	/// let property = MutableProperty(0)
-	/// let signal = Signal({ /* do some work after some time */ })
-	/// property <~ signal
-	/// ````
-	///
-	/// ````
-	/// let property = MutableProperty(0)
-	/// let signal = Signal({ /* do some work after some time */ })
-	/// let disposable = property <~ signal
-	/// ...
-	/// // Terminates binding before property dealloc or signal's
-	/// // `completed` event.
-	/// disposable.dispose()
-	/// ````
-	///
-	/// - parameters:
-	///   - target: A target to be bond to.
-	///   - signal: A signal to bind.
-	///
-	/// - returns: A disposable that can be used to terminate binding before the
-	///            deinitialization of the target or the signal's `completed`
-	///            event.
+	associatedtype Error: Swift.Error
```

Comment:
This would need an error type until Swift 4 since we can't add conditional conformance when `Error == NoError`.